packages/core/src/anchor/anchor-processing-loop.ts

```diff
@@ -0,0 +1,64 @@
+import { ProcessingLoop } from './processing-loop.js'
+import type { CASClient } from './anchor-service.js'
+import type {
+  AnchorRequestStore,
+  AnchorRequestStoreListResult,
+} from '../store/anchor-request-store.js'
+import { AnchorRequestCarFileReader } from './anchor-request-car-file-reader.js'
+import type { AnchorLoopHandler } from './anchor-service.js'
+import type { DiagnosticsLogger } from '@ceramicnetwork/common'
+import type { NamedTaskQueue } from '../state-management/named-task-queue.js'
+
+/**
+ * Get anchor request entries from AnchorRequestStore one by one. For each entry, get CAS response,
+ * and handle the response via `eventHandler.handle`.
+ *
+ * If CAS does not know about an anchor request, we consider it being not created previously for some reason.
+ * We create a request on CAS then.
+ */
+export class AnchorProcessingLoop {
+  readonly #loop: ProcessingLoop<AnchorRequestStoreListResult>
+  readonly #queue: NamedTaskQueue
+
+  constructor(
+    batchSize: number,
+    cas: CASClient,
+    store: AnchorRequestStore,
+    logger: DiagnosticsLogger,
+    eventHandler: AnchorLoopHandler,
+    queue: NamedTaskQueue
+  ) {
+    this.#queue = queue
+    this.#loop = new ProcessingLoop(store.infiniteList(batchSize), (entry) =>
+      this.#queue.run(entry.key.toString(), async () => {
```

Comment:
what's the advantage of using the NamedTaskQueue here?  It just serializes handling for a single streamid, but we're already guaranteed that everything coming out of the AnchorRequestStore is going to be for a unique StreamID since StreamID is the key in the store.

Can we just get rid of `#queue` entirely?

Reply:
Aha!  I now see we use the same queue in the EthereumAnchorService to serialize writes to the AnchorRequestStore there.  So now I see that the goal of this is to serialize all writes to the AnchorRequestStore, instead of using the ExecutionQueue for that.  Nice.  Can we just add some comments to the queue both here and in the EthereumAnchorService to make the clearer?  And maybe also rename it in both places to something like `anchorStoreQueue` or something?

Reply:
wow, the deeper I look at things like `Repository.handleAnchorEvent` the more I realize how awesome and elegant this approach is.  It totally lets us work around not having a transactional state store while *almost* giving us proper transactional read-modify-write semantics, all without slowing down regular write processing since it all happens outside of the ExecutionQueue.

It's so close to being a perfect solution, but there is still one small race condition I believe.  The actual query to the AnchorRequestStore doesn't happen within the queue, so it's possible that by the time the `onValue` handler in the ProcessingLoop is run, the entry in the AnchorRequestStore could have changed from what we read out and polled the CAS for.  That could cause us to delete an entry for a newer tip from the store because we hit a terminal state with an older tip that we read out.

Can you think of any ways to close that final gap?  The only way I can think of would be to change the `store.infiniteList` batch query to just get the keys from the store but not the values, and then within the actual `onValue` processor load the value from the AnchorRequestStore so that you're already within the queue and are guaranteed that the value can't then change out from under you.  It's more i/o to the store which I don't love, but it would fully solve all the race conditions around managing the store once and for all.  What do you think?

Reply:
A game of tradeoffs again. Considering AnchorRequestStore is most probably a local thing, I guess, IO overhead is not that big a deal.

Reply:
Yeah, I think it is done.